hardware/tftp/Android.mk

```diff
@@ -1,36 +1,29 @@
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/rfs/shared rfs/msm/mpss/shared && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/hlos_rfs/shared rfs/msm/mpss/hlos && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /firmware rfs/msm/mpss/readonly/firmware && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /odm/firmware rfs/msm/mpss/readonly/vendor/firmware && popd > /dev/null)
+TARGET_COMBINATIONS := \
+    /data/vendor/tombstones/modem:ramdumps \
+    /persist/rfs/msm/mpss:readwrite \
+    /persist/rfs/shared:shared \
+    /persist/hlos_rfs/shared:hlos \
+    /firmware:readonly/firmware \
+    /odm/firmware:readonly/vendor/firmware
 
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /data/vendor/tombstones/lpass rfs/msm/adsp/ramdumps && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/rfs/msm/adsp rfs/msm/adsp/readwrite && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/rfs/shared rfs/msm/adsp/shared && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/hlos_rfs/shared rfs/msm/adsp/hlos && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /firmware rfs/msm/adsp/readonly/firmware && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /odm/firmware rfs/msm/adsp/readonly/vendor/firmware && popd > /dev/null)
+TARGET_PREFIXES := \
+    rfs/msm/mpss \
+    rfs/msm/adsp \
+    rfs/msm/slpi \
+    rfs/msm/cdsp
 
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /data/vendor/tombstones/rfs/slpi rfs/msm/slpi/ramdumps && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/rfs/msm/slpi rfs/msm/slpi/readwrite && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/rfs/shared rfs/msm/slpi/shared && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /persist/hlos_rfs/shared rfs/msm/slpi/hlos && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /firmware rfs/msm/slpi/readonly/firmware && popd > /dev/null)
-$(shell pushd $(PRODUCT_OUT)/$(TARGET_COPY_OUT_VENDOR) > /dev/null && ln -sf /odm/firmware rfs/msm/slpi/readonly/vendor/firmware && popd > /dev/null)
+# Prepend vendor and prefix directory to all link names:
+SONY_SYMLINKS := $(foreach prefix,$(TARGET_PREFIXES), \
+    $(foreach s,$(TARGET_COMBINATIONS), \
+        $(eval p := $(subst :,$(space),$(s))) \
+        $(word 1,$(p)):$(TARGET_COPY_OUT_VENDOR)/$(prefix)/$(word 2,$(p)) \
+    ) \
+)
```

Comment:
this is fairly complex.

writing a verbose list of the form `t1:n1 t2:n2 ...` here might be better since it easier to maintain.

The advantage of just keeping a flat list of the form  is that it is very

Reply:
As explained in the commit description, this is done intentionally. From what it looks these are all the same, and converting it to an outer product enforces that. Reducing typos at the same time.

However, you may be right in that I had to add an edgecase for the tombstones which do not follow this pattern (accidentally?). And the case does not hold for the `readwrite` symlinks. Again, this may be intentional, but it could easily be the typo this code is preventing against:

(Diff comparing current `HEAD` to the new symlink creation, generated with `find out/target/product/akatsuki/vendor -type l -exec echo -n {}" -> " \; -exec readlink {} \; | sort -u`)
```diff --git a/old_symlinks b/new2_symlinks
index c2f3085..d53c970 100644
--- a/old_symlinks
+++ b/new2_symlinks
@@ -205,13 +205,13 @@ out/target/product/akatsuki/vendor/rfs/msm/adsp/hlos -> /persist/hlos_rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/adsp/ramdumps -> /data/vendor/tombstones/lpass
 out/target/product/akatsuki/vendor/rfs/msm/adsp/readonly/firmware -> /firmware
 out/target/product/akatsuki/vendor/rfs/msm/adsp/readonly/vendor/firmware -> /odm/firmware
-out/target/product/akatsuki/vendor/rfs/msm/adsp/readwrite -> /persist/rfs/msm/adsp
+out/target/product/akatsuki/vendor/rfs/msm/adsp/readwrite -> /persist/rfs/msm/mpss
 out/target/product/akatsuki/vendor/rfs/msm/adsp/shared -> /persist/rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/cdsp/hlos -> /persist/hlos_rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/cdsp/ramdumps -> /data/vendor/tombstones/rfs/cdsp
 out/target/product/akatsuki/vendor/rfs/msm/cdsp/readonly/firmware -> /firmware
 out/target/product/akatsuki/vendor/rfs/msm/cdsp/readonly/vendor/firmware -> /odm/firmware
-out/target/product/akatsuki/vendor/rfs/msm/cdsp/readwrite -> /persist/rfs/msm/cdsp
+out/target/product/akatsuki/vendor/rfs/msm/cdsp/readwrite -> /persist/rfs/msm/mpss
 out/target/product/akatsuki/vendor/rfs/msm/cdsp/shared -> /persist/rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/mpss/hlos -> /persist/hlos_rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/mpss/ramdumps -> /data/vendor/tombstones/modem
@@ -223,5 +223,5 @@ out/target/product/akatsuki/vendor/rfs/msm/slpi/hlos -> /persist/hlos_rfs/shared
 out/target/product/akatsuki/vendor/rfs/msm/slpi/ramdumps -> /data/vendor/tombstones/rfs/slpi
 out/target/product/akatsuki/vendor/rfs/msm/slpi/readonly/firmware -> /firmware
 out/target/product/akatsuki/vendor/rfs/msm/slpi/readonly/vendor/firmware -> /odm/firmware
-out/target/product/akatsuki/vendor/rfs/msm/slpi/readwrite -> /persist/rfs/msm/slpi
+out/target/product/akatsuki/vendor/rfs/msm/slpi/readwrite -> /persist/rfs/msm/mpss
 out/target/product/akatsuki/vendor/rfs/msm/slpi/shared -> /persist/rfs/shared```

@jerpelea Is this correct, or should all `*/readwrite` links point to their own `/persist/rfs/msm/*` folder?